check50/__main__.py

```diff
@@ -203,7 +206,55 @@ def raise_invalid_slug(slug, offline=False):
         msg += _("\nIf you are confident the slug is correct and you have an internet connection," \
                 " try running without --offline.")
 
-    raise internal.Error(msg)
+    raise _exceptions.Error(msg)
+
+
+def process_args(args):
+    """Validate arguments and apply defaults that are dependent on other arguments"""
+
+    # dev implies offline, verbose, and log level "INFO" if not overwritten
+    if args.dev:
+        args.offline = True
+        args.verbose = True
+        if "ansi" in args.output:
+            args.ansi_log = True
+
+    # offline implies local
+    if args.offline:
+        args.no_install_dependencies = True
+        args.no_download_checks = True
+        args.local = True
+
+    if not args.log_level:
+        args.log_level = LogLevel.WARNING
+
+    # Setup logging for lib50
+    setup_logging(args.log_level)
+
+    # Warning in case of running remotely with no_download_checks or no_install_dependencies set
+    if not args.local:
+        useless_args = []
+        if args.no_download_checks:
+            useless_args.append("--no-downloads-checks")
+        if args.no_install_dependencies:
+            useless_args.append("--no-install-dependencies")
+
+        if useless_args:
+            LOGGER.warning(_("You should always use --local when using: {}").format(", ".join(useless_args)))
+
+    # Filter out any duplicates from args.output
+    seen_output = []
+    for output in args.output:
+        if output in seen_output:
+            LOGGER.warning(_("Duplicate output format specified: {}").format(output))
+        else:
+            seen_output.append(output)
+
+    args.output = seen_output
```

Comment:
Should we just use a `set` out of `args.output` and use it? We could still show a generic warning that duplicate output formats are ignored if the size of the set is less than the size of the list if we want to but I don't think it's a big deal.

Reply:
The original thinking here was that order mattered. If you supplied: `-o ansi json html`, check50 would then deliver output in that order. That's why it's implemented in the way it is.

But, I might go back on this because I'm not entirely sure this level of control is at all needed. For now, no reason to change though.